content/en/flux/guides/helmreleases.md

```diff
@@ -385,6 +385,131 @@ removed from the cluster by kustomize-controller if
 [pruning](../components/kustomize/kustomization/#garbage-collection) is enabled.
 {{% /alert %}}
 
+## Refer to values in Secret generated with Kustomize and SOPS
+
+It is possible to use Kustomize [Secret generator](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/secretgenerator/)
+to trigger a Helm release upgrade every time the encrypted secret values change.
+
+A SOPS configuration for your cluster is required first. Follow the [Manage Kubernetes secrets with Mozilla SOPS](../mozilla-sops/)
+guide. The details of configuring SOPS are out of scope for this entry.
+
+Once you have SOPS configured, create a `kustomizeconfig.yaml` for Kustomize to
+be able to patch `Secrets` referenced in `HelmRelease` manifests:
+
+```yaml
+nameReference:
+- kind: Secret
+  version: v1
+  fieldSpecs:
+  - path: spec/valuesFrom/name
+    kind: HelmRelease
+```
+
+Create a `HelmRelease` definition that references a `Secret`:
+
+```yaml
+apiVersion: helm.toolkit.fluxcd.io/v2beta1
+kind: HelmRelease
+metadata:
+  name: podinfo
+  namespace: podinfo
+spec:
+  interval: 5m
+  releaseName: podinfo
+  chart:
+    spec:
+      chart: podinfo
+      sourceRef:
+        kind: HelmRepository
+        name: podinfo
+  valuesFrom:
+    - kind: Secret
+      name: podinfo-values
+```
+
+Ensure that this HelmRelease will be applied from the same Flux Kustomization
+that will decrypt the values secret. The encrypted secret data must be housed
+in the same Flux Kustomization path as the `HelmRelease` in order to allow
+Kustomize's Secret generator function to compose them together with the
+Kustomize configuration.
+
+Create a `kustomization.yaml` that generates the `Secret`:
+
+```yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: podinfo
+resources:
+  - namespace.yaml
+  - repository.yaml
+  - release.yaml
+secretGenerator:
+  - name: podinfo-values
+    files:
+      - values.yaml=my-values.enc.yaml
+configurations:
+  - kustomizeconfig.yaml
+```
+
+Now `HelmRelease` values can come from an encrypted secret! Read on below to
+prepare the secret file and apply it to the cluster in a Flux Kustomization.
+
+#### Encrypting a `values.yaml` with the SOPS CLI
+
+To prepare the encrypted secret, please note there are some divergences from
+encrypting a well-formed Kubernetes Secret with metadata and versioning.
+
+Run this command to encrypt to the output file that was named in
+`kustomization.yaml` above:
+```
+$ sops -e --input-type=yaml --output-type=yaml my-values.tmp.secret > my-values.enc.yaml
+```
+
+Commit the `my-values.enc.yaml` file and discard the temp file, being sure not
+to accidentally commit secrets to the repository. Read on for more information.
+
+If users have followed the [SOPS guide](../mozilla-sops/) then we likely added a `creation_rules`
+entry telling the `sops` CLI how to handle encrypting all YAML secrets, like:
+
+```yaml
+# ./apps/sensitive/.sops.yaml
+creation_rules:
+  - path_regex: ".*\.yaml"
```

Comment:
This rule does not match `my-values.tmp.secret` so how come is an issue?

Reply:
Some explanation is definitely necessary because the failure mode is very bad. If people take `values.yaml` and try to encrypt it directly, it will come out completely unencrypted with the configuration that is put forward in the SOPS guide, unless it had a section `data` or `stringData`. Plus I really didn't want to write "don't call your values.yaml file `values.yaml`" it just sounds wrong. I really wanted to include stronger language and a more specific method that doesn't add any risk the unencrypted file will accidentally get committed before it gets encrypted.

I actually wanted to mention in a separate document something like the unencrypted secret precommit hook that prevents you from committing and pushing a secret that hasn't been encrypted, I stumbled upon the other day. But it necessarily depends on the secrets being well-formed Kubernetes secrets to detect and block them from being committed. I have no mitigation strategy to prevent folks from accidentally committing a values.yaml that hasn't been encrypted properly, except to warn people against mixing secrets and non-secrets, and advising directly to use some caution when handling secrets.

Reply:
Let's use `values.yaml` please